src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java

```diff
@@ -35,8 +40,12 @@ public AuthenticationResponse register(final RegisterRequest registerRequest)
 				.updated(registerRequest.getUpdated())
 				.build();
 		userRepository.save(user);
-		var jwt = jwtService.generateToken(user);
-		return AuthenticationResponse.builder().token(jwt).build();
+		verificationService.createOTPAndSaveInDB(user);
+		eventPublisher.publishEvent(String.valueOf(user.getId()));
+		return RegisterResponse.builder()
+				.userId(user.getId())
+				.status(user.getStatus().getStatus())
```

Comment:
I think this part will be a reason for NPE. Please check

Reply:
****